### PR TITLE
JAMES-2301 JMS mail queue do not preserve per-recipient headers [CONTRIB]

### PR DESCRIPTION
**Pull request by chibenwa**
CI for https://github.com/apache/james-project/pull/123

### CODE REVIEW

File: server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSMailQueue.java [hunk 1]
```diff
@@ -297,27 +303,20 @@ protected void produceMail(Map<String, Object> props, int msgPrio, Mail mail) th
...
-            String recipient = recipients.next().toString();
-            recipientsBuilder.append(recipient.trim());
-            if (recipients.hasNext()) {
-                recipientsBuilder.append(JAMES_MAIL_SEPARATOR);
-            }
+        // won't serialize the empty headers so it is mandatory
+        // to handle nulls when reconstructing mail from message
+        if (!mail.getPerRecipientSpecificHeaders().getHeadersByRecipient().isEmpty()) {
+            byte[] serialize = SerializationUtils.serialize(mail.getPerRecipientSpecificHeaders());
+            props.put(JAMES_MAIL_PER_RECIPIENT_HEADERS, Hex.encodeHexString(serialize));
```
Comment (rouazana) [inline server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSMailQueue.java]: why not base64?

Comment (rouazana) [reply server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSMailQueue.java]: thanks, cherry-picked

**Comment by chibenwa** [comment]
test this please (build stopped)

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**